packages/support/lib/image-util.js

```diff
@@ -293,23 +265,24 @@ async function getImagesMatches (img1Data, img2Data, options = {}) {
                     `Only ${JSON.stringify(AVAILABLE_MATCHING_FUNCTIONS)} matching functions are supported.`);
   }
 
-  const detector = new cv[`${detectorName}Detector`]();
+  const detector = new cv[detectorName]();
   const [img1, img2] = await B.all([
-    cv.imdecodeAsync(img1Data),
-    cv.imdecodeAsync(img2Data)
-  ]);
-  const [result1, result2] = await B.all([
-    detectAndCompute(img1, detector),
-    detectAndCompute(img2, detector)
+    cvMatFromImage(img1Data),
+    cvMatFromImage(img2Data)
   ]);
+  const result1 = detectAndCompute(img1, detector);
+  const result2 = detectAndCompute(img2, detector);
+  const matcher = new cv.DescriptorMatcher(matchFunc);
+  const matchesVec = new cv.DMatchVector();
   let matches = [];
-  try {
-    matches = await cv[`match${matchFunc}Async`](result1.descriptor, result2.descriptor);
-  } catch (e) {
-    throw new Error(`Cannot find any matches between the given images. Try another detection algorithm. ` +
-                    ` Original error: ${e}`);
+  matcher.match(result1.descriptor, result2.descriptor, matchesVec);
+  const totalCount = matchesVec.size();
+  if (totalCount < 1) {
+    throw new Error(`Could not find any matches`);
```

Comment:
Would it be possible to have an error message here, that contains some hint for a user on how to solve this error?

Reply:
yea